src/app/config/projectProperties.json

```diff
@@ -2,6 +2,7 @@
     "projectTypes": {
         "taskpane": {
             "displayname": "Office Add-in Task Pane project",
+            "manifestType": "xml",
```

Comment:
Instead of specifying the manifest type you should specify the manifest relative path.  That way is a single lookup without any if statements that have special information about subfolders for specific project types.